starlark/etcd_histogram.go

```diff
@@ -0,0 +1,72 @@
+package starlark
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/vmware-tanzu/crash-diagnostics/ssh"
+	"go.starlark.net/starlark"
+)
+
+// etcdHistogramFn is the Starlark built-in for the fetching kubernetes objects
+// and returns the result as a Starlark value containing the file path and error message, if any
+// Starlark format: kube_capture(what="logs" [, groups="core", namespaces=["default"], kube_config=kube_config()])
+func etcdHistogramFn(thread *starlark.Thread, _ *starlark.Builtin, args starlark.Tuple, kwargs []starlark.Tuple) (starlark.Value, error) {
+	var workdir, fileName, desc string
+	var metric string
+	var resources *starlark.List
+
+	if err := starlark.UnpackArgs(
+		identifiers.capture, args, kwargs,
+		"resources?", &resources,
+		"workdir?", &workdir,
+		"file_name?", &fileName,
+		"desc?", &desc,
+		"metric", &metric,
+	); err != nil {
+		return starlark.None, fmt.Errorf("%s: %s", identifiers.capture, err)
+	}
+	if len(workdir) == 0 {
+		if dir, err := getWorkdirFromThread(thread); err == nil {
+			workdir = dir
+		}
+	}
+	if len(workdir) == 0 {
+		workdir = defaults.workdir
+	}
+
+	if resources == nil {
+		res, err := getResourcesFromThread(thread)
+		if err != nil {
+			return starlark.None, fmt.Errorf("%s: %s", identifiers.copyFrom, err)
+		}
+		resources = res
+	}
+
+	var agent ssh.Agent
+	var ok bool
+	if agentVal := thread.Local(identifiers.sshAgent); agentVal != nil {
+		agent, ok = agentVal.(ssh.Agent)
+		if !ok {
+			return starlark.None, errors.New("unable to fetch ssh-agent")
+		}
+	}
+
+	cmd := "sudo curl -sk --cert /etc/kubernetes/pki/apiserver-etcd-client.crt --key /etc/kubernetes/pki/apiserver-etcd-client.key https://localhost:2379/metrics"
```

Comment:
We would need to make sure the `curl` command is available in the images.

Reply:
thats true, @vladimirvivien does prereq work for the resources as well as the executing workstation?